src/tree-sitter-language-mode.js

```diff
@@ -1337,7 +1337,26 @@ class NodeCursorAdaptor {
   }
 }
 
-class NullHighlightIterator {
+class NullLanguageModeHighlightIterator {
+  seek() {
+    return [];
+  }
+  compare() {
+    return 1;
+  }
+  moveToSuccessor() {}
+  getPosition() {
+    return Point.INFINITY;
+  }
+  getOpenScopeIds() {
+    return [];
+  }
+  getCloseScopeIds() {
+    return [];
+  }
+}
```

Comment:
i do not understand the fix, but i assume that if you want to fix it and avoid code duplication, you can also do:

```js
class NullLanguageModeHighlightIterator extends NullLayerHighlightIterator {
seek() {
    return [];
  }
}
```

And should work the same.

Reply:
Sure. Could also use mixins or whathaveyou. However, the actual (non-null) iterators are not actually related beyond sharing method names, and are used entirely separately (i.e. not intended to be interchangeable in any meaningful way). So I'm a little apprehensive about making null- iterators dependent on each other's implementation. It might create similar issues going forward.

Reply:
So at this point in the list `possible drawback` the code duplication is not a drawback at all if the code is intended to behave differently.

Reply:
I'm on the fence about this. On one hand, there _is_ apparent code duplication. On another hand, null iterators mimic non-null iterators structurally (i.e. both null and non-null iterators use separate implementations). Non-null iterators behave very differently and share very little code beyond method names. Null iterators are pretty similar however, but arguably that's a coincidence, and a shared implementation is what basically caused #22078 in the first place.

Overall, I'm leaning to "there's not so much code duplication as to really worry about it, and there are potential drawbacks to reducing it".

I should've been more clear about the reason for doing it the way I did in the OP, but I was in a bit of a hurry when I made this PR.